lib/vector-core/src/event/metric/value.rs

```diff
@@ -597,14 +600,49 @@ impl ByteSizeOf for Sample {
     }
 }
 
+/// Custom serialization function which converts special `f64` values to strings.
+/// Non-special values are serialized as numbers.
+#[allow(clippy::trivially_copy_pass_by_ref)]
+fn serialize_f64<S>(value: &f64, serializer: S) -> Result<S::Ok, S::Error>
+where
+    S: Serializer,
+{
+    if value.is_infinite() || value.is_nan() {
+        serializer.serialize_str(&format!("{value}"))
```

Comment:
This ends up allocating a string just to drop it. If we are concerned about performance, this could be broken down to avoid the allocation:
```suggestion
    if value.is_infinite() {
        serializer.serialize_str(if value > 0 { "inf" } else { "-inf" })
    } else if value.is_nan() {
        serializer.serialize_str("NaN")
```